### PR TITLE
Ui/key bindings

### PR DESCRIPTION
**Pull request by odedre**
## Proposed Changes

- added new key bindings
- created base menu component
- added key bindings to component menu 
- added key bindings to scope and workspace menu and using base menu
- created command bar context

### CODE REVIEW

File: scopes/ui-foundation/uis/main-dropdown/main-dropdown.tsx [hunk 1]
```diff
@@ -25,42 +41,44 @@ export function MainDropdown() {
...
+function Line({ lineTitle, keyChar, onClick, ...rest }: LineProps) {
   return (
-    <div className={classNames(hoverable, styles.line)}>
-      <div>{title}</div>
-      <Icon of={of} className={styles.key} />
+    <div {...rest} className={classNames(styles.line)} onClick={onClick}>
+      <div>{lineTitle}</div>
+      {keyChar && (
+        <pre>
+          <KeyCombo>{keyChar}</KeyCombo>
```
Comment (KutnerUri) [inline scopes/ui-foundation/uis/main-dropdown/main-dropdown.tsx:79]: please use `<KeySequence>` (= combo + sequence)

File: scopes/workspace/workspace/workspace.ui.runtime.tsx [hunk 2]
```diff
@@ -57,23 +57,15 @@ export class WorkspaceUI {
     return this;
   }
 
-  private registerExplicitRoutes() {
```
Comment (KutnerUri) [inline scopes/workspace/workspace/workspace.ui.runtime.tsx:60]: where is this??

File: scopes/scope/scope/scope.ui.runtime.tsx [hunk 3]
```diff
@@ -59,9 +53,16 @@ export class ScopeUI {
...
-    private menuWidgetSlot: MenuWidgetSlot
+    private menuWidgetSlot: MenuWidgetSlot,
+
+    /**
+     * main dropdown item slot
+     */
+    private menuItemSlot: MenuItemSlot
   ) {}
 
+  private setKeyBindHandler: (updated: CommandHandler) => void = () => {};
```
Comment (KutnerUri) [inline scopes/scope/scope/scope.ui.runtime.tsx:64]: let's talk about this again

**Review by KutnerUri — CHANGES_REQUESTED** [review]

File: scopes/workspace/workspace/workspace.ui.runtime.tsx (hunk 2 above)
Comment (odedre) [reply scopes/workspace/workspace/workspace.ui.runtime.tsx:60]: moved to the provider like ran asked

**Comment by KutnerUri** [comment]
also consider - 

* rename ui.main-dropdown -> ellipsis-dropdown / harmburger-dropdown
* remove unused command-bar-context

File: scopes/component/component/component.ui.runtime.tsx [hunk 4]
```diff
@@ -55,28 +55,30 @@ export class ComponentUI {
...
+    }
+  };
+
   /**
    * key bindings used by component aspect
    */
   private keyBindings: CommandEntry[] = [
     {
-      id: 'copyBitId', // extract to constant!
+      id: 'copyBitId', // TODO - extract to a component!
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:72]: 1. prefix with `'component.'`
1. extract to constant! you can move it to another file in this same component

File: scopes/component/component/component.ui.runtime.tsx [hunk 5]
```diff
@@ -55,28 +55,30 @@ export class ComponentUI {
...
-      id: 'copyNpmId', // extract to constant!
-      handler: () => {
-        const packageName = this.activeComponent?.packageName;
-        if (packageName) {
-          const version = this.activeComponent?.id.version;
-          const versionString = version ? `@${version}` : '';
-          copy(`${packageName}${versionString}`);
-        }
-      },
+      id: 'copyNpmId', // TODO - extract to a component!
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:80]: same

**Review by KutnerUri — CHANGES_REQUESTED** [review]

**Review by KutnerUri — APPROVED** [review]

**Pull request merged**